### PR TITLE
Add image build config and release script

### LINKED ISSUE #6: Push not to local registy
Currently the build is configured in the `pom.xml` that it pushes to a local registry:
https://github.com/knative-extensions/eventing-integrations/blob/main/pom.xml#L28-L29

However for other java dataplane bits we have adjusted the build process. For instance:
* https://github.com/knative-extensions/eventing-kafka-broker/blob/b9233feecbfdadb1c4a584d3e9925cc117b2427c/hack/data-plane.sh#L59

### PR DESCRIPTION
**Pull request by dsimansk**
# Changes

- Add image build config and release script

In a nutshell, this will be driven by the same `release.sh` as rest of the components. I'll create a nightly Prow job to verify it first. 

IMO we could have at least another `kind` profile or some local one for development, but let's see.

Fixes #6 

/cc @matzew @christophd

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -199,6 +199,23 @@
...
+            <properties>
+                <quarkus.container-image.image>${env.KO_DOCKER_REPO}/${project.artifactId}:${env.TAG}</quarkus.container-image.image>
+
+                <quarkus.container-image.build>true</quarkus.container-image.build>
+                <quarkus.container-image.push>false</quarkus.container-image.push>
+
+                <!-- Unset to prevent default inheritance -->
+                <quarkus.container-image.insecure>false</quarkus.container-image.insecure>
+            </properties>
+        </profile>
```
Comment (matzew) [inline pom.xml:218]: In addition, you think we should generally leave it to local-horst?
https://github.com/knative-extensions/eventing-integrations/blob/5d47d4604ee99fc272797acb53bcd6a469207e51/pom.xml#L28

I guess, in theory, I'd like no hard-coded `localhost`, but something that resolves via ENV_VAR? 

Here is a "big fan" of `$KO_DOCKER_REPO` like we do on `golang` :upside_down_face: 

/cc @christophd

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module knative.dev/eventing-integrations
+
+go 1.23.1
+
+require knative.dev/hack v0.0.0-20241010131451-05b2fb30cb4d
```
Comment (dsimansk) [inline go.mod:5]: We only need `go.mod` to enable our hack scripts.

Comment (matzew) [reply go.mod:5]: sounds good, @dsimansk 

Thanks a lot for getting into this

File: pom.xml (hunk 1 above)
Comment (christophd) [reply pom.xml:218]: "local-horst" 🤣 

I am not tied to using this local config as a default. I can adjust the settings on my fork when testing locally. Feel free to remove and set a decent default value like quay.io or something

Comment (dsimansk) [reply pom.xml:218]: I wanted to isolate changes for now, but sure it make sense to use `${env.KO_DOCKER_REPO}`. A note here, quarkus doesn't like additional namespaces in `registry` property, only `host:port` combination. So `kind.local` is fine, but not `gcr.io/knative-nightly` - our default in release script. :)

Therefore I've used whole `image` property here.

**Comment by dsimansk** [comment]
Nightly Prow job to test it out and publish the images: https://github.com/knative/infra/pull/532

File: pom.xml [hunk 3]
```diff
@@ -199,6 +199,23 @@
...
     <profiles>
+        <profile>
+            <id>knative</id>
+            <activation>
+                <property>
+                    <name>knative</name>
+                </property>
+            </activation>
+            <properties>
+                <quarkus.container-image.image>${env.KO_DOCKER_REPO}/${project.artifactId}:${env.TAG}</quarkus.container-image.image>
```
Comment (dsimansk) [inline pom.xml:210]: @christophd @matzew currently the module names by directory and artifactId are different. E.g. `  <artifactId>kn-connector-aws-s3-source</artifactId>` for aws-s3-sink. Should we align those, or create a name property to be reflected in the image name.

Comment (dsimansk) [reply pom.xml:210]: I see Eventing core is expecting kn prefixed name. That's fine for now.

```
if source.Spec.Aws.S3 != nil {
			return "quay.io/openshift-knative/kn-connector-aws-s3-source:1.0-SNAPSHOT"
		}
```

Comment (matzew) [reply pom.xml:210]: :see_no_evil: 

let's fix that :rofl:

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**